server/get_payload.go

```diff
@@ -92,35 +124,116 @@ func processPayload[P Payload](m *BoostService, log *logrus.Entry, ua UserAgent,
-			if err != nil {
-				if errors.Is(requestCtx.Err(), context.Canceled) {
-					// This is expected if the payload has already been received by another relay
-					log.Info("request was cancelled")
-				} else {
-					log.WithError(err).Error("error making request to relay")
+			// If the request is in SSZ but the relay indicated with getHeader that it does not
+			// support SSZ, we must convert the request blinded beacon block from SSZ to JSON.
+			// Another option would be to move this outside of the loop and do the conversion
+			// once if any relays do not support SSZ. The problem with this though is that it
+			// would penalize relays which do support SSZ, whereas this version only penalizes
+			// relays which do not support SSZ.
+			if parsedProposerContentType == MediaTypeOctetStream && !relay.SupportsSSZ {
+				log.Info("converting request from SSZ to JSON for relay")
+				signedBlindedBeaconBlockBytes, err = convertSSZToJSON(proposerEthConsensusVersion, signedBlindedBeaconBlockBytes)
+				if err != nil {
+					log.WithError(errFailedToConvert).Error("failed to convert SSZ to JSON")
+					return
 				}
+			}
+
+			// If the request fails, try again a few times with 100ms between tries
+			resp, err := retry(requestCtx, m.requestMaxRetries, 100*time.Millisecond, func() (*http.Response, error) {
+				// Make a new request
+				req, err := http.NewRequestWithContext(requestCtx, http.MethodPost, url, bytes.NewReader(signedBlindedBeaconBlockBytes))
+				if err != nil {
+					log.WithError(err).Warn("error creating new request")
+					return nil, err
+				}
+
+				// Add header fields to this request
+				req.Header.Set(HeaderAccept, proposerAcceptContentTypes)
+				req.Header.Set(HeaderContentType, proposerContentType)
+				req.Header.Set(HeaderEthConsensusVersion, proposerEthConsensusVersion)
+				req.Header.Set(HeaderKeySlotUID, currentSlotUID)
+				req.Header.Set(HeaderStartTimeUnixMS, fmt.Sprintf("%d", time.Now().UTC().UnixMilli()))
+				req.Header.Set(HeaderUserAgent, userAgent)
+
+				// Send the request
+				log.Debug("requesting payload")
+				resp, err := m.httpClientGetPayload.Do(req)
+				if err != nil {
+					log.WithError(err).Warn("error calling getPayload on relay")
+					return nil, err
+				}
+
+				// Check that the response was successful
+				if resp.StatusCode != http.StatusOK {
+					err = fmt.Errorf("%w: %d", errHTTPErrorResponse, resp.StatusCode)
+					log.WithError(err).Warn("error status code")
+					return nil, err
+				}
+
+				return resp, nil
+			})
+			if err != nil {
+				log.WithError(err).Warn("failed to get payload from relay after retries")
+				return
+			}
+			defer resp.Body.Close()
+
+			// Get the resp body content
+			respBytes, err := io.ReadAll(resp.Body)
+			if err != nil {
+				log.WithError(err).Warn("error reading response body")
+				return
+			}
+
+			// Get the response's content type
+			respContentType := resp.Header.Get(HeaderContentType)
```

Comment:
do all relays properly set this header? might be safer to fallback to JSON for now if not set

Reply:
Good call. I agree we should default to JSON here. Will make this change.

Reply:
Ah wait. We actually already do this but it's in the `getPayload` function:

https://github.com/flashbots/mev-boost/blob/d7df45be70ffb0228cb4986e7c373efabe4abff8/server/get_payload.go#L49-L54

Reply:
maybe I got confused, thought the code snippet you linked is parsing content type of proposer (ie. CL client calling mev-boost) and the line I commented on is the response from relay/builder to the `submitBlindedBlock` call

Reply:
Nope. You were right & I was the confused one. Sorry about that.

Fixed here: https://github.com/flashbots/mev-boost/pull/742/commits/5fa6b03fda12a47b3e448243a4a3e38256fb4bb7